### ISSUE TITLE
ResourceExhausted error from gRPC: `grpc: received message after decompression larger than max`

### ISSUE THREAD

**Issue by astencel-sumo** [SumoLogic/sumologic-kubernetes-collection#2576]
K8s collection version: `v2.17.0`

# Steps to reproduce:

1. Install the collection chart in a cluster with traces enabled:

    ```yaml
    sumologic:
      acessId: xxx
      accessKey: yyy
      traces:
        enabled: true
    ```

2. Send a batch of spans that is larger than 4 MB.

## Expected result:

- The spans are successfully forwarded to Sumo Logic

## Actual result:

- An error similar to the following is emitted from the `otelagent` instance that received the batch (this error actually comes from the `metrics` pipeline, but it would be the same for the `traces` pipeline):

    "2022-10-07T14:40:21.537Z\ error\ exporterhelper/queued_retry.go:183\ Exporting failed. The error is not retryable. Dropping data.\ {\"kind\": \"exporter\", \"data_type\": \"metrics\", \"name\": \"otlp\", \"error\": \"Permanent error: rpc error: code = ResourceExhausted desc = grpc: received message after decompression larger than max (4243263 vs. 4194304)\", \"dropped_items\": 11866}\

# Cause of the error

The traces pipeline looks like the following (the naming isn't the most friendly):

```
 end-user   ||      Sumo k8s collection chart      ||    Sumo backend
application => otelagent => otelgateway => otelcol => Sumo Logic backend
```

The `otelcol` instances have the `batch` processor's `send_batch_max_size` property set to `512`, which prevents Sumo Logic from complaining that the request is too big. But the `otelagent` and `otelgateway` do not have the `send_batch_max_size` property set. This means that a big batch of spans (or a big batch of metrics) sent to the `otelagent` will be sent "as-is" to `otelgateway` and then to `otelcol` via OTLP/gRPC. The gRPC protocol has a default server setting of `4 MB` for the [maximum message size](https://pkg.go.dev/google.golang.org/grpc?utm_source=godoc#MaxRecvMsgSize). This causes the `otlp` exporter in the `otelagent` instance to receive a  gRPC error like the above from the `otlp` receiver in the `otelgateway` instance.

# How to fix

## Option 1: Increase the `max_recv_msg_size_mib` option on the OTLP receiver on `otelgateway` and `otelcol`

By doing this, we don't have to split big batches in `otelagent` or `otelgateway`, this can be left for the last stage after any optional filtering/sampling of spans.

The small problem with this approach is that any value we set will be arbitrary.

## Option 2: Set the `send_batch_max_size` option on the `batch` processor in `otelagent` and `otelgateway`

By doing this, we split the big batches sooner in the pipeline. For simplicity, we can just duplicate the value of the `send_batch_max_size` from the `otelcol` instances (currently `512`).

The problem with the approach is, if the user has aggressive sampling enabled in `otelcol`, it might occur that the job is being done twice for no benefit: first we split big batches in the beginning of the pipeline, but then in `otelcol` we might need to merge those batches again after sampling.

**Comment by sumo-drosiek** [comment]
> but then in otelcol we might need to merge those batches again after sampling.

I think this is task for backend and not for otelcol

> The small problem with this approach is that any value we set will be arbitrary.
What are risks of setting this value to some very high value, e.g. 50GB?

**Comment by astencel-sumo** [comment]
Fixed in #2839
